### PR TITLE
Throw error for unsupported scenarios in dotnet-ef

### PR DESCRIPTION
**Pull request by natemcmaster**
Point users to a webpage with dotnet-ef RC2 limitations when:

1. When dotnet-ef fails on a class library project or
2. Loading external startup on desktop .NET

Fix https://github.com/aspnet/EntityFramework/issues/5321

### CODE REVIEW

**Comment by natemcmaster** [comment]
cc @divega

**Comment by divega** [comment]
LGTM. I think @rowanmiller should review the message and provide the forward link.

**Comment by natemcmaster** [comment]
@divega I've updated this PR to include the removal of "--startup-project". :cry:

File: src/Microsoft.EntityFrameworkCore.Tools/tools/EntityFramework.psm1 [hunk 1]
```diff
@@ -921,6 +921,11 @@ function GetProjectItem($project, $path) {
 
 function GetStartUpProject($name, $fallbackProject) {
     if ($name) {
+        if (IsDotNetProject $fallbackProject) {
+            # this means users specified -StartupProject explicitly
+            # otherwise, ignore what VS has marked as the "startup project"
+            Write-Warning "'-StartupProject' is not supported on .NET Core projects. This parameter will be ignored."
```
Comment (divega) [inline src/Microsoft.EntityFrameworkCore.Tools/tools/EntityFramework.psm1]: Is it not supported in .NET Core projects or in project.json based projects?

Comment (natemcmaster) [inline src/Microsoft.EntityFrameworkCore.Tools/tools/EntityFramework.psm1]: Good catch. It's all "xproj" type projects.

**Comment by divega** [comment]
I had a question about an error message in the PowerShell commands but everything else LGTM.

**Comment by natemcmaster** [comment]
Also added a fix for #4832 because this is going to be encountered much more now.

File: src/Microsoft.EntityFrameworkCore.Tools/DispatchCommand.cs [hunk 2]
```diff
@@ -123,10 +137,20 @@ public static CommandLineApplication Create()
...
                 {
-                    Reporter.Verbose.WriteLine(ex.Message);
-                    // intentionally put DispatcherToolName in error because "Microsoft.EntityFrameworkCore.Tools.Cli" is 
-                    // brought in automatically as a dependency of "Microsoft.EntityFrameworkCore.Tools"
-                    Reporter.Error.WriteLine(ToolsStrings.ProjectDependencyCommandNotFound(DispatcherToolName));
+                    if (isClassLibrary)
+                    {
+                        // TODO
+                        var fwlink = "";
+                        Reporter.Error.WriteLine($"This version of Entity Framework tools for .NET Core CLI does not support class library projects. See {fwlink} for more details.");
```
Comment (rowanmiller) [inline src/Microsoft.EntityFrameworkCore.Tools/DispatchCommand.cs]: http://go.microsoft.com/fwlink/?LinkId=798221

Here is my suggestion for the message...

> This preview of Entity Framework tools does not support targeting class library projects in ASP.NET Core and .NET Core applications. See {fwlink} for details and workarounds.

File: src/Microsoft.EntityFrameworkCore.Tools/tools/EntityFramework.psm1 [hunk 3]
```diff
@@ -921,6 +921,11 @@ function GetProjectItem($project, $path) {
 
 function GetStartUpProject($name, $fallbackProject) {
     if ($name) {
+        if (IsDotNetProject $fallbackProject) {
+            # this means users specified -StartupProject explicitly
+            # otherwise, ignore what VS has marked as the "startup project"
+            Write-Warning "'-StartupProject' is not supported on .NET Core projects (*.xproj). This parameter will be ignored."
```
Comment (natemcmaster) [inline src/Microsoft.EntityFrameworkCore.Tools/tools/EntityFramework.psm1]: @rowanmiller here's another error message to look at

**Comment by natemcmaster** [comment]
Updated with PR feedback.

File: src/Microsoft.EntityFrameworkCore.Tools/tools/EntityFramework.psm1 (hunk 3 above)
Comment (divega) [inline src/Microsoft.EntityFrameworkCore.Tools/tools/EntityFramework.psm1]: I think we should remove the reference to ".NET Core" in the error completely.

File: src/Microsoft.EntityFrameworkCore.Tools.Core/Properties/ToolsCoreStrings.resx [hunk 4]
```diff
@@ -237,4 +237,7 @@ Change your target project to the migrations project by using the Package Manage
   <data name="ForceRemoveMigration" xml:space="preserve">
     <value>Removing migration '{name}' without checking the database. If this migration has been applied to the database, you will need to manually reverse the changes it made.</value>
   </data>
+  <data name="NoParameterlessConstructor" xml:space="preserve">
+    <value>No parameterless constructor on type '{contextType}' was available. Either add one to '{contextType}' or add an implementation of 'IDbContextFactory&lt;{contextType}&gt;'.</value>
```
Comment (divega) [inline src/Microsoft.EntityFrameworkCore.Tools.Core/Properties/ToolsCoreStrings.resx]: @rowanmiller new exception message if I am not mistaken.

**Comment by divega** [comment]
:shipit: with any necessary changes to exceptions.

File: src/Microsoft.EntityFrameworkCore.Tools.Core/Properties/ToolsCoreStrings.resx (hunk 4 above)
Comment (rowanmiller) [inline src/Microsoft.EntityFrameworkCore.Tools.Core/Properties/ToolsCoreStrings.resx]: Minor changes....

> No parameterless constructor was found on '{contextType}'. Either add a parameterless constructor to '{contextType}' or add an implementation of 'IDbContextFactory&lt;{contextType}&gt;' in the same assembly as '{contextType}'.

**Comment by rowanmiller** [comment]
:shipit: with exception message updates

**Pull request merged**